### ISSUE TITLE
ConnectionRefusedError won't close connection

### ISSUE THREAD

**Issue by youyueyinfeng** [miguelgrinberg/python-socketio#491]
_handle_eio_connect() return the value of _handle_connect(), but the _handle_connect() dosen't return anything. 

It seems that _handle_disconnect() returned a value before commit 805d5f37413a1e3bbad22012237412803217b4b9, which commit info is 'Support the connect_error event in the client'. In this comimt, 'return False' was deleted. This made client connect_error worked, but made ConnectionRefusedError failed as well.

**Comment by miguelgrinberg** [comment]
Do you have an actual problem with the current versions of the engineio and socketio packages? If you have an application that does not do what is expected please show it to me so that I can debug the problem and fix it. The area of the code you are referring to has gone through several changes lately, but I'm not aware of any issues at this time.

**Comment by youyueyinfeng** [comment]
For example, at server side I want to accept connections with some authentication, maybe a header X-CLIENT-ID. If there is no X-CLIENT-ID, rejects it.
A demo client.py:
```
import socketio
sio = socketio.Client()
sio.connect(
       'http://localhost:5000',
        # headers={'X-CLIENT-ID': 'e1df2f5a-2062-4634-813c-61abb72a3491'}
)
sio.wait()
```
A demo server.py
```
import socketio
from socketio import exceptions

sio = socketio.Server()
@sio.connect
def connect(sid, environ):
       if not environ.get('HTTP_X_CLIENT_ID'):
              raise exceptions.ConnectionRefusedError('Authentication failed')
app = socketio.WSGIApp(sio)

from gevent import pywsgi
from geventwebsocket.handler import WebSocketHandler
from geventwebsocket import WebSocketServer, WebSocketApplication
pywsgi.WSGIServer(('0.0.0.0', 5000), app, handler_class=WebSocketHandler).serve_forever()
```
If _handle_connect() return False, server will return a 401 response when ConnectionRefusedError raises, and client will exit for receving an unexpected status code. But in current version, client will connect successfully. Perhaps there's another realization for this scenery?

**Comment by miguelgrinberg** [comment]
> But in current version, client will connect successfully.

Not for me.

```
(venv) $ python client.py
Traceback (most recent call last):
  File "ccc.py", line 4, in <module>
    'http://localhost:5000',
  File "/Users/mgrinberg/Documents/dev/python/python-socketio/socketio/client.py", line 282, in connect
    six.raise_from(exceptions.ConnectionError(exc.args[0]), None)
  File "<string>", line 3, in raise_from
socketio.exceptions.ConnectionError: Unexpected status code 401 in server response
```

If I uncomment the header in the client then I do get a connection.

**Comment by youyueyinfeng** [comment]
Oh sorry, it is my mistake. I git cloned the code at 5.17. However, the code was updated and fixed this at 5.22 in commit 8d08096dc442e817b5e4ce53321ccf196daafcd1. Thank you for helping me.
